libraries/tickfs/src/flash_controller.rs

```diff
@@ -0,0 +1,79 @@
+//! The Flash Controller interface with hardware
```

Comment:
What is the relationship between a FlashController and an allocated in-kernel storage region made with the `storage_volume` macro? Does this represent a controller over a logical region (e.g., a storage region) or the physical entire flash? It seems like TickFS should operate over a storage region -- is that how it's intended to operate?